### PR TITLE
freeze the pipeline commit for all stages in a pipeline run

### PR DESCRIPTION
**Pull request by boris-dimitrov**
freeze the pipeline commit for all stages in a pipeline run

### CODE REVIEW

File: app/models/pipeline_run.rb [hunk 1]
```diff
@@ -34,8 +34,6 @@ class PipelineRun < ApplicationRecord
   STATUS_READY = 'READY'.freeze
   POSTPROCESS_STATUS_LOADED = 'LOADED'.freeze
 
-  before_create :create_run_stages
```
Comment (yunfangjuan) [inline app/models/pipeline_run.rb]: did you do a scan on PipelineRun to make sure there's only one place for create_run_stages?

Comment (boris-dimitrov) [reply app/models/pipeline_run.rb]: i did

but i now realize i don't need to change this

**Comment by boris-dimitrov** [comment]
Yeah valid point about the fact that if there is a bug in the pipeline code a simple retry would not recover.   However, only admins can fix such a bug, and they have the power to retry with the fix (aka rerun vs resume).

**Comment by boris-dimitrov** [comment]
I think we may be able to do better in choosing when to retry/resume.  I think resume is useful only when the failed stage is stage 2.  Basically resume=lazy.

**Pull request merged**